### PR TITLE
Update to reusable workflows

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
# What does this PR do?
Updates github workflows to be reusable.
Idea is to make these changes to all 4 infra collection repos, so that when we make a change, it is automatically reflected in all 4 repositories. Currently there is a Issue to create a repo to store it in redhat_cop, for now its the repo I setup.
Adds sanity tests
Adds tests before releases and revamps release and changelog workflows
Fixes errors in lookup plugin discovered by sanity tests.

# How should this be tested?
Automated tests
Results here
Release with attachments
https://github.com/sean-m-sullivan/controller_configuration/releases/tag/1.0.11
Release Workflow
https://github.com/sean-m-sullivan/controller_configuration/actions/runs/3617309300

### CODE REVIEW

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -1,25 +1,35 @@
 ---
 # This workflow action will run pre-commit, which will execute ansible and yaml linting
 # See .pre-commit-config.yaml for what hooks are executed
-name: Yaml and Ansible Lint
+name: Sanity and Linting
```
Comment (Tompage1994) [inline .github/workflows/pre-commit.yml]: Probably shouldn't go with 'sanity' and rather use 'confidence' or something like that here.

File: changelogs/fragments/workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
```
Comment (Tompage1994) [inline changelogs/fragments/workflow.yaml]: Probably don't need this . I think the changelog should be kept to functionality changes

File: .github/workflows/pre-commit.yml [hunk 3]
```diff
@@ -1,25 +1,35 @@
...
+
+on:
+  pull_request:
+  push:
+  schedule:
+    - cron: "0 6 * * *"
 
 jobs:
   pre-commit:
+    uses: sean-m-sullivan/infra_git_ci_cd/.github/workflows/pre-commit.yml@main
```
Comment (Tompage1994) [inline .github/workflows/pre-commit.yml:15]: Not sure I like this being under your namespace. Any way we can get this under redhat-cop?

**Review by Tompage1994 — CHANGES_REQUESTED** [review]

File: .github/workflows/pre-commit.yml (hunk 3 above)
Comment (sean-m-sullivan) [reply .github/workflows/pre-commit.yml:15]: https://github.com/redhat-cop/org/issues/537
this is temporary until that is created

File: changelogs/fragments/workflow.yaml (hunk 2 above)
Comment (sean-m-sullivan) [reply changelogs/fragments/workflow.yaml]: I can remove that, it was put in mainly to test the changelog workflow changes.

File: .github/workflows/pre-commit.yml (hunk 1 above)
Comment (sean-m-sullivan) [reply .github/workflows/pre-commit.yml]: Changed

**Review by Tompage1994 — APPROVED** [review]

**Review by djdanielsson — APPROVED** [review]
I need to checkout the action itself at some point but the changes here seem fine

**Pull request merged**